src/libexpr/nixexpr.hh

```diff
@@ -430,9 +449,45 @@ MakeBinOp(ExprOpNEq, "!=")
 MakeBinOp(ExprOpAnd, "&&")
 MakeBinOp(ExprOpOr, "||")
 MakeBinOp(ExprOpImpl, "->")
-MakeBinOp(ExprOpUpdate, "//")
 MakeBinOp(ExprOpConcatLists, "++")
 
+struct ExprOpUpdate : Expr
+{
+    PosIdx pos;
+    Expr *e1, *e2;
+    ExprOpUpdate(Expr * e1, Expr * e2)
+        : e1(e1)
+        , e2(e2){};
+    ExprOpUpdate(const PosIdx & pos, Expr * e1, Expr * e2)
+        : pos(pos)
+        , e1(e1)
+        , e2(e2){};
+    void show(const SymbolTable & symbols, std ::ostream & str) const override
+    {
+        str << "(";
+        e1->show(symbols, str);
+        str << " "
+               "//"
+               " ";
+        e2->show(symbols, str);
+        str << ")";
+    }
+    void bindVars(EvalState & es, const std ::shared_ptr<const StaticEnv> & env) override
+    {
+        e1->bindVars(es, env);
+        e2->bindVars(es, env);
+    }
+    void eval(EvalState & state, Env & env, Value & v) override;
+    PosIdx getPos() const override
+    {
+        return pos;
+    }
+
+    virtual void evalForUpdate(EvalState & state, Env & env, UpdateQueue & q) override;
```

Comment:
This is the only new code in this declaration. The rest is just an expansion of the `MakeBinOp` macro.

Reply:
Should make a `MakeBinOpMembers` macro that `MakeBinOp` this can both uses?